### PR TITLE
Aj 1037 select filtered

### PR DESCRIPTION
**Pull request by calypsomatic**
### Jira Ticket: [AJ-1037](https://broadworkbench.atlassian.net/browse/AJ-1037)

Using the 'exact match filter' on a column was not propagating correctly to either selecting rows or exporting.  Thus, if a user used exact match filter and then chose 'select all filtered' and tried to download to a tsv, the result would be the entier entity set instead of the filtered set as expected.  This was due to the filter not being included as a parameter when determining which entities were on the page/selected.

### What
- Adds the 'columnFilter' parameter to `paginatedEntitiesOfType`
- Adds unit tests around selecting rows and copying filtered selections
- Fixes a pre-existing problem with `HeaderCell` that caused ` Warning: Each child in a list should have a unique "key" prop`

### Testing strategy
- [x] Manually testing
- [x] Added unit tests

 

[AJ-1037]: https://broadworkbench.atlassian.net/browse/AJ-1037?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/data/EntitiesContent.js [hunk 1]
```diff
@@ -284,9 +284,8 @@ const EntitiesContent = ({
   } = useColumnProvenance(workspace, entityKey);
   const [showColumnProvenance, setShowColumnProvenance] = useState(undefined);
```
Comment (calypsomatic) [inline src/components/data/EntitiesContent.js]: `EntitiesContent` is here because I branched off of https://github.com/DataBiosphere/terra-ui/pull/4126 ; please ignore it for now.

File: src/components/data/DataTable.test.ts [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+
+    // Should include all rows + the 'Select all' check
+    const allChecks = screen.getAllByRole('checkbox');
+    expect(allChecks.length).toEqual(101);
+
+    // Assert
+    // They should all be checked
+    allChecks.forEach((checkbox) => {
+      expect(checkbox).toBeChecked();
+    });
```
Comment (nawatts) [inline src/components/data/DataTable.test.ts]: Suggest combining these into one assertion. Something like:
```ts
const numCheckboxesChecked = allChecks.filter(el => el.checked).length;
expect(numCheckbxoesChecked).toBe(101);
```

File: src/components/data/EntitiesContent.test.ts [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+    await user.click(exportButton);
+    const copyMenuItem = screen.getByRole('menuitem', { name: 'Copy to clipboard' });
+    const copyButton = within(copyMenuItem).getByRole('button');
+    await user.click(copyButton);
+
+    // Assert
+    expect(clipboard.writeText).toHaveBeenCalledWith('entity:sample_set_id\nsample_set_1\n');
+  });
+
+  it('copies filtered table to clipboard', async () => {
```
Comment (calypsomatic) [inline src/components/data/EntitiesContent.test.ts:232]: This is the only test added for this PR; the rest is a carryover from https://github.com/DataBiosphere/terra-ui/pull/4126

**Review by nawatts — APPROVED** [review]

File: src/components/data/DataTable.js [hunk 4]
```diff
@@ -211,7 +211,8 @@ const DataTable = (props) => {
     });
 
   const getAllEntities = async () => {
-    const params = _.pickBy(_.trim, { pageSize: filteredCount, filterTerms: activeTextFilter, filterOperator });
+    const colFilt = !!columnFilter.filterColAttr && !!columnFilter.filterColTerm ? `${columnFilter.filterColAttr}=${columnFilter.filterColTerm}` : '';
```
Comment (jladieu) [inline src/components/data/DataTable.js]: Maybe extract a method for the right-hand side of the assignment?  I notice the same logic on line 179.  This also might make space to expand the variable `colFilt` to something a bit more descriptive like 'columnFilterQueryString`.

File: src/components/data/DataTable.test.ts [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+
+  return {
+    ...actual,
+    AutoSizer: MockAutoSizer,
+  };
+});
+
+const getPage = jest
+  .fn()
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (jladieu) [inline src/components/data/DataTable.test.ts:56]: Just curious what was complaining before you put in this lint disable?

File: src/components/data/DataTable.test.ts [hunk 6]
```diff
@@ -0,0 +1,424 @@
...
+  .mockImplementation((signal, entityType, queryOptions: { pageNumber; columnFilter }, entityMetadata) => {
```
Comment (jladieu) [inline src/components/data/DataTable.test.ts:57]: Consider adding brief comment explaining the intended behavior of the mock?  (eg: pretends to be X pages with Y items)

File: src/components/table.js [hunk 7]
```diff
@@ -905,8 +905,8 @@ export const TooltipCell = ({ children, tooltip, ...props }) =>
     [h(TextCell, props, [children])]
   );
 
-export const HeaderCell = (props) => {
-  return h(TextCell, _.merge({ style: { fontWeight: 600 } }, props));
+export const HeaderCell = ({ children, ...props }) => {
```
Comment (jladieu) [inline src/components/table.js:908]: Is this the fix for the 3rd bullet you mention in the PR description?  This one might worth be splitting out into its own PR with an explanation, as it seems kinda random when placed alongside the rest of the PR :)

**Review by jladieu — APPROVED** [review]
Amazing job improving the test coverage here.  It looks like it was a ton of work to get everything set up to be testable, but I appreciate how you make it safer for our future selves.

I left a few optional suggestions for code organization and readability, but looks really good to me!

File: src/components/data/DataTable.test.ts (hunk 5 above)
Comment (calypsomatic) [reply src/components/data/DataTable.test.ts:56]: I only need the `queryOptions` parameter for this mock but I need to include the correct number of parameters in the correct order for it to work - but eslint doesn't like the parameters that I'm not using
